tekton/integration_pipeline_test.go

```diff
@@ -330,6 +332,41 @@ var _ = Describe("Integration pipeline", func() {
 			Expect(string(enterpriseContractPipelineRun.Spec.PipelineRef.ResolverRef.Resolver)).To(Equal("git"))
 		})
 
+		It("can add timeouts to the IntegrationPipelineRun according to the environment variables", func() {
+			// The pipelineRun timeouts should be empty if environment vars are not set
+			newIntegrationPipelineRun.WithDefaultIntegrationTimeouts()
+
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Pipeline).To(BeNil())
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Tasks).To(BeNil())
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Finally).To(BeNil())
+
+			// Set the timeouts to invalid strings, which should skip setting the timeouts
+			os.Setenv("PIPELINE_TIMEOUT", "thisIsNotAValidDuration!")
+			os.Setenv("TASKS_TIMEOUT", "thisIsNotAValidDuration!")
+			os.Setenv("FINALLY_TIMEOUT", "thisIsNotAValidDuration!")
+			newIntegrationPipelineRun.WithDefaultIntegrationTimeouts()
+
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Pipeline).To(BeNil())
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Tasks).To(BeNil())
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Finally).To(BeNil())
+
+			// This should work and the pipelineRun spec should have the values
+			os.Setenv("PIPELINE_TIMEOUT", "2h")
+			os.Setenv("TASKS_TIMEOUT", "2h")
+			os.Setenv("FINALLY_TIMEOUT", "2h")
+			expectedDuration, _ := time.ParseDuration("2h")
+			newIntegrationPipelineRun.WithDefaultIntegrationTimeouts()
+
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Pipeline.Duration).To(Equal(expectedDuration))
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Tasks.Duration).To(Equal(expectedDuration))
+			Expect(newIntegrationPipelineRun.Spec.Timeouts.Finally.Duration).To(Equal(expectedDuration))
+
+			// Reset the environment variables
+			os.Setenv("PIPELINE_TIMEOUT", "")
```

Comment:
This should be reworked to proper Before/AfterEach, or `defer` so we don't have envvars messing around when a Expect fails in this test case

Reply:
Good catch, moved the envvars setting to Before and AfterEach.